optuna/distributions.py

```diff
@@ -207,8 +211,8 @@ class DiscreteUniformDistribution(BaseDistribution):
             A discretization step.
     """
 
-    def __init__(self, low, high, q):
-        # type: (float, float, float) -> None
+    def __init__(self, low: float, high: float, q: float) -> None:
+        high = _adjust_discrete_uniform_high(low, high, q)
```

Comment:
How about adjusting and raising a warning after the `low > high` check? I think this gives the user a slightly better experience as the error message becomes more clear with the unmodified high and the warning is skipped altogether. We can probably simplify some unit tests as well. 

This comment also applies to the following distribution classes.

Reply:
> How about adjusting and raising a warning after the `low > high` check?

Thank you for your suggestion. I agree with you. It will help users to resolve their issues.
Fixed: https://github.com/optuna/optuna/pull/1276/commits/5798c297aea3e92d67d6aa70d2a5574b2d1edea0

> We can probably simplify some unit tests as well.

You're right. I removed unnecessary filters of `UserWarning` in the commit https://github.com/optuna/optuna/pull/1276/commits/a88cbb0c5fbe8eb54f0ccb613ac8c8f515d0c0eb.